pkg/flowaggregator/flowaggregator.go

```diff
@@ -468,47 +468,67 @@ func (fa *flowAggregator) sendTemplateSet(isIPv6 bool) (int, error) {
 		if err != nil {
 			return 0, fmt.Errorf("%s not present. returned error: %v", ie, err)
 		}
-		ie := ipfixentities.NewInfoElementWithValue(element, nil)
+		ie, err := ipfixentities.DecodeAndCreateInfoElementWithValue(element, nil)
+		if err != nil {
+			return 0, err
+		}
 		elements = append(elements, ie)
 	}
 	for _, ie := range ianaReverseInfoElements {
 		element, err := fa.registry.GetInfoElement(ie, ipfixregistry.IANAReversedEnterpriseID)
 		if err != nil {
 			return 0, fmt.Errorf("%s not present. returned error: %v", ie, err)
 		}
-		ie := ipfixentities.NewInfoElementWithValue(element, nil)
+		ie, err := ipfixentities.DecodeAndCreateInfoElementWithValue(element, nil)
+		if err != nil {
+			return 0, err
+		}
 		elements = append(elements, ie)
 	}
 	for _, ie := range antreaInfoElements {
 		element, err := fa.registry.GetInfoElement(ie, ipfixregistry.AntreaEnterpriseID)
 		if err != nil {
 			return 0, fmt.Errorf("%s not present. returned error: %v", ie, err)
 		}
-		ie := ipfixentities.NewInfoElementWithValue(element, nil)
+		ie, err := ipfixentities.DecodeAndCreateInfoElementWithValue(element, nil)
+		if err != nil {
+			return 0, err
+		}
 		elements = append(elements, ie)
 	}
-	for _, ie := range antreaSourceStatsElementList {
-		element, err := fa.registry.GetInfoElement(ie, ipfixregistry.AntreaEnterpriseID)
+	for i := range antreaSourceStatsElementList {
```

Comment:
Lets use `antreaStatsElementList` to be consistent with go-ipfix code? In addition, please add a comment that the order has to match go-ipfix code indicating the function name. Next time, it might be easy to debug this.

Reply:
Sure. Updated.